### PR TITLE
Get root for invocation target

### PR DESCRIPTION
**Pull request by aljones15**
Adds a function that takes an `invocationTarget` and returns a the `expectedRootCapability` for it.
This is only used in the `_verifyDelegation` function which is used by `routes.authorizations`.

I can replace all `expectedRootCapability` calls with this if wanted, but it seems like overkill.

So let me know.

Also controller is being passed to `_verifyDelegation`, but is not used and is not an object apparently.

### CODE REVIEW

File: lib/http.js [hunk 1]
```diff
@@ -748,6 +750,69 @@ async function _generateRootCapability(url) {
...
+    }
+    // chunks and documents by id have the same rootCap
+    if(verifyDocPath(invocationTarget)) {
+      return [
+        `${edvId}/zcaps${docPath}`,
+        `${edvId}/zcaps/documents`
+      ];
+    }
+  }
+  if(Array.isArray(invocationTarget)) {
```
Comment (dlongley) [inline lib/http.js]: `invocationTarget` should not be an array. We should remove this ... and see out if it's ever being called like that and figure out what's going on.

File: lib/http.js [hunk 2]
```diff
@@ -748,6 +750,69 @@ async function _generateRootCapability(url) {
...
+ *   invocationTarget.
+ * @param {string} options.edvId - The edvId.
+ * @param {string} [options.docPath] - An optional docPath.
+ *
+ * @return {string|Array<string>} The expectedRootCapability for
+ *   the invocationTarget.
+ */
+function _getExpectedRootCapability({invocationTarget = '', edvId, docPath}) {
+  const verifyDocPath = url =>
+    url.includes('documents') && !url.endsWith('documents');
```
Comment (dlongley) [inline lib/http.js]: We should move this out to a separate helper function so we don't create a new function each time this is called.

Comment (dlongley) [reply lib/http.js]: Also, based on the code path below, once the `invocationTarget` is an array path is removed, we won't need to check for `url.endsWith(documents)` anymore. It seems like we can just check to see if the target ends with `/documents/${docPath}` ... though, looking at where this gets called, `docPath` is never passed. Seems like there's more to do here.

**Comment by aljones15** [comment]
I like the precision of the new code and I managed to create a new instance using it so I say good to go.

**Pull request merged**